commcare_export/commcare_hq_client.py

```diff
@@ -30,6 +32,12 @@
 
 def on_backoff(details):
     _log_backoff(details, 'Waiting for retry.')
+    response = details["exception"].response
+    if response.status_code == 429:
+        retry_after = response.headers.get("Retry-After", 0.0)
+        retry_after = ceil(float(retry_after))
+        logger.warning(f"Sleeping for {retry_after} seconds")
+        time.sleep(retry_after)
```

Comment:
this isn't what I had in mind but I guess it works.

Reply:
Another way is to use the `wait_gen` parameter to give it the time to wait for, but it would need to be context aware. Currently the decorator is not. Do do this, we can wrap this whole get method along with the decorator inside an instance method. That way we can get a `wait_gen` generator that is able to determine how long to wait for, depending on the Retry-After value.

Reply:
Yea, I was thinking of something like that. It might get ugly though. We'd also need to modify the `response.raise_for_status()` call so that we don't do double backoff.

I'm fine with this for now if you would rather try the other approach later.

Reply:
Thoughts on this approach? https://github.com/dimagi/commcare-export/pull/225/commits/388a5077df9bada11af0ebf29f6ff28080d1ebe1

Reply:
Looks good to me